packages/expo-doctor/src/utils/files.ts

```diff
@@ -2,27 +2,37 @@ import spawnAsync from '@expo/spawn-async';
 import fs from 'fs';
 import path from 'path';
 
+import { isFileIgnoredByRules } from './isFileIgnoredByRules';
+
+const EASIGNORE_FILENAME = '.easignore';
+const GITIGNORE_FILENAME = '.gitignore';
+
 /**
- * Checks if a file exists and is not ignored by git.
+ * Checks if a file exists and is not ignored by EAS (.easignore) or git (.gitignore).
  * @param filePath The path to the file to check.
- * @returns `true` if the file exists and is not ignored by git.
+ * @returns `true` if the file exists and is not ignored.
  */
 export async function existsAndIsNotIgnoredAsync(filePath: string): Promise<boolean> {
   return fs.existsSync(filePath) && !(await isFileIgnoredAsync(filePath));
 }
 
 /**
- * Checks if a file is ignored by git.
+ * Checks if a file is ignored by EAS (.easignore) or git (.gitignore).
+ * Prioritizes .easignore if it exists, otherwise falls back to .gitignore.
  * @param filePath The path to the file to check.
- * @returns `true` if the file is ignored by git.
+ * @returns `true` if the file is ignored, `false` otherwise.
  */
 export async function isFileIgnoredAsync(filePath: string): Promise<boolean> {
-  try {
-    await spawnAsync('git', ['check-ignore', '-q', filePath], {
-      cwd: path.normalize(await getRootPathAsync()),
-    });
-    return true;
-  } catch {
+  const rootPath = await getRootPathAsync();
+  const easIgnorePath = path.join(rootPath, EASIGNORE_FILENAME);
+  const gitIgnorePath = path.join(rootPath, GITIGNORE_FILENAME);
+
+  if (fs.existsSync(easIgnorePath)) {
+    return isFileIgnoredByRules(filePath, easIgnorePath, rootPath);
+  } else if (fs.existsSync(gitIgnorePath)) {
+    return isFileIgnoredByRules(filePath, gitIgnorePath, rootPath);
```

Comment:
I thought gitignore files work on their subfolders, even if that subfolder has another gitignore, and you would want the gitignore to still work even if it wasn't in the project root folder. Therefore, I think this logic might need to fall back to `git` when an easignore file isn't present. https://stackoverflow.com/a/3307252

Reply:
You're right! I'll update the else statement to use our previous logic using git